src/quantuminspire/api.py

```diff
@@ -614,22 +648,23 @@ def get_raw_data_from_result(self, result_id: int) -> List[int]:
         except IndexError as err_msg:
             raise ApiError(f'Invalid raw data url for result with id {result_id}!') from err_msg
         try:
-            raw_data: List[int] = self._action(['results', 'raw-data', 'read'], params={'id': result_id,
-                                                                                        'token': token})
+            raw_data: List[List[Any]] = self._action(['results', 'raw-data', 'read'],
+                                                     params={'id': result_id,
+                                                             'token': token})
         except ErrorMessage as err_msg:
             raise ApiError(f'Raw data for result with id {result_id} does not exist!') from err_msg
         return raw_data
 
-    def get_quantum_states_from_result(self, result_id: int) -> List[Any]:
-        """ Gets the quantum states from the result of the executed job, given the result_id.
+    def get_quantum_states_from_result(self, result_id: int) -> List[List[Any]]:
+        """ Gets the quantum states for each measurement block from the result of the executed job, given the result_id.
 
         :param result_id: The identification number of the result.
 
         :raises ApiError: If the quantum states url in result is invalid or the request
             for the quantum states using the url failed.
 
         :return:
-            The quantum states consists of a list of quantum state values.
+            A list of quantum states. The quantum states consists of a list of quantum state values.
```

Comment:
The vagueness is there because none of our backends currently return quantum states (always empty list).
I can image is consists of a list of state values and their corresponding probability (I added something like that to the docstring)